pkg/landscaper/controllers/execution/controller.go

```diff
@@ -143,7 +160,13 @@ func HandleAnnotationsAndGeneration(ctx context.Context, log logr.Logger, c clie
 func HandleErrorFunc(log logr.Logger, client client.Client, eventRecorder record.EventRecorder, exec *lsv1alpha1.Execution) func(ctx context.Context, err error) error {
 	old := exec.DeepCopy()
 	return func(ctx context.Context, err error) error {
-		exec.Status.LastError = lserrors.TryUpdateError(exec.Status.LastError, err)
+		if reflect.DeepEqual(old.Status.LastError, exec.Status.LastError) && err == nil {
```

Comment:
with that we save compute if the err is not nil
```suggestion
		if err == nil && reflect.DeepEqual(old.Status.LastError, exec.Status.LastError)  {
```